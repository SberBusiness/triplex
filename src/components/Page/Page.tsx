import React from 'react';
import {classnames} from '@sberbusiness/triplex/utils/classnames/classnames';
import {BodyPage} from '@sberbusiness/triplex/components/Page/components/BodyPage';
import {FooterPage} from '@sberbusiness/triplex/components/Page/components/FooterPage';
import {HeaderPage} from '@sberbusiness/triplex/components/Page/components/HeaderPage';

/** Свойства компонента Page. */
interface IPageProps extends React.HTMLAttributes<HTMLDivElement> {}

interface IPageFC extends React.FC<IPageProps> {
    Header: typeof HeaderPage;
    Body: typeof BodyPage;
    Footer: typeof FooterPage;
}

/** Страница. Может содержать только PageHeader, BodyPage, PageFooter. */
export const Page: IPageFC = ({children, className, ...htmlDivAttributes}) => (
    <div className={classnames('cssClass[page]', className)} {...htmlDivAttributes}>
        {children}
    </div>
);

Page.Header = HeaderPage;
Page.Body = BodyPage;
Page.Footer = FooterPage;
