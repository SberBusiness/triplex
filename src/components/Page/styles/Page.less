@import 'src/styles/style';

.page {
    height: 100%;
    color: @ufs-theme-default-text-color;
    display: flex;
    flex-grow: 1; // Для IE11
    flex-direction: column;
}

.bodyPage {
    flex-grow: 1;
}
