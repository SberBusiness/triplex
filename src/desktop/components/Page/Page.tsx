import * as React from 'react';
import {classnames} from '@sberbusiness/triplex/common/utils/classnames/classnames';
import {BodyPage} from '@sberbusiness/triplex/desktop/components/Page/components/BodyPage';
import {FooterPage} from '@sberbusiness/triplex/desktop/components/Page/components/FooterPage';
import {HeaderPage} from '@sberbusiness/triplex/desktop/components/Page/components/HeaderPage';

interface IPageProps extends React.HTMLAttributes<HTMLDivElement> {}

interface IPageFC extends React.FC<IPageProps> {
    Header: typeof HeaderPage;
    Body: typeof BodyPage;
    Footer: typeof FooterPage;
}

/**
 * Компонент страница. Может содержать только PageHeader, BodyPage, PageFooter.
 */
export const Page: IPageFC = ({children, className, ...htmlDivAttributes}) => (
    <div className={classnames('cssClass[page]', className)} {...htmlDivAttributes}>
        {children}
    </div>
);

Page.Header = HeaderPage;
Page.Body = BodyPage;
Page.Footer = FooterPage;
