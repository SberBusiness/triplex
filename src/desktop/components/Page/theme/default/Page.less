@import 'src/desktop/common/theme/default';

.page {
    height: 100%;
    color: @ufs-theme-default-text-color;
    display: flex;
    flex-grow: 1; // Для IE11
    flex-direction: column;
}

.bodyPage {
    flex-grow: 1;
}
