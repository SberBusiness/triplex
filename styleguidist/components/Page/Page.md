```jsx
import React from 'react';
import {Button} from '@sberbusiness/triplex/components/Button/Button';
import {EButtonTheme, EButtonSize} from '@sberbusiness/triplex/components/Button/enums';

<div style={{height: 500}}>
    <Page>
        <Page.Header>
            <Page.Header.Link href="#" onClick={(event) => event.preventDefault()}>
                Текст ссылки
            </Page.Header.Link>
            <Page.Header.Title>
                <Page.Header.Title.Content>
                    <Page.Header.Title.Content.Text>
                        Шаблонный текст заголовка в одну строку
                    </Page.Header.Title.Content.Text>
                    <Page.Header.Title.Content.Subhead>
                        Шаблонный текст для описания
                    </Page.Header.Title.Content.Subhead>
                </Page.Header.Title.Content>
                <Page.Header.Title.Controls>
                    <Button theme={EButtonTheme.GENERAL} size={EButtonSize.MD}>
                        Button Name
                    </Button>
                    <Button theme={EButtonTheme.SECONDARY} size={EButtonSize.MD}>
                        Button Name
                    </Button>
                </Page.Header.Title.Controls>
            </Page.Header.Title>
        </Page.Header>
        <Page.Body>
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
            labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco
            laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in
            voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat
            cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </Page.Body>
        <Page.Footer>
            <Page.Footer.Description>
                <Page.Footer.Description.Content>Текст в одну строку</Page.Footer.Description.Content>
                <Page.Footer.Description.Controls>
                    <Button theme={EButtonTheme.GENERAL} size={EButtonSize.MD}>
                        Button Name
                    </Button>
                    <Button theme={EButtonTheme.SECONDARY} size={EButtonSize.MD}>
                        Button Name
                    </Button>
                </Page.Footer.Description.Controls>
            </Page.Footer.Description>
        </Page.Footer>
    </Page>
</div>
```
