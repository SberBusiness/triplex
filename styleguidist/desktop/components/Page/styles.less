.page-example {
    display: flex;
    height: 500px;
}
