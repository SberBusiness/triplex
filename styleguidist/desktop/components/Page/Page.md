```jsx
import React, {useState} from 'react';
import {Button} from '@sberbusiness/triplex/desktop/components/Button/Button';
import {EButtonTheme, EButtonSize} from '@sberbusiness/triplex/desktop/components/Button/enums';
import {Page} from '@sberbusiness/triplex/desktop/components/Page/Page';
import {Row} from '@sberbusiness/triplex/desktop/components/Row/Row';
import {Col} from '@sberbusiness/triplex/desktop/components/Col/Col';
import {Gap} from '@sberbusiness/triplex/desktop/components/Gap/Gap';
import {EBodyBackgroundColor} from '@sberbusiness/triplex/desktop/components/Body/enums';
import './styles.less';

const [backgroundColor, setBackgroundColor] = useState(EBodyBackgroundColor.WHITE);

const renderSelect = (label, value, handler, options) => (
    <label style={{display: 'inline-flex'}}>
        {label}
        <select value={value} onChange={(event) => handler(event.target.value)} style={{marginLeft: '4px'}}>
            {options.map((value, index) => (
                <option key={index} value={value}>
                  {value}
                </option>
            ))}
        </select>
    </label>
);

<>
    {renderSelect('Background Color', backgroundColor, setBackgroundColor, Object.values(EBodyBackgroundColor))}
    <Gap size={16} />
    <div className="page-example">
        <Page>
            <Page.Header>
                <Page.Header.Link>Текст ссылки</Page.Header.Link>
                <Page.Header.Title>
                    <Page.Header.Title.Content>
                        <Page.Header.Title.Content.Text>Шаблонный текст заголовка в одну строку</Page.Header.Title.Content.Text>
                        <Page.Header.Title.Content.Subhead>Шаблонный текст для описания</Page.Header.Title.Content.Subhead>
                    </Page.Header.Title.Content>
                    <Page.Header.Title.Controls>
                        <Button theme={EButtonTheme.GENERAL} size={EButtonSize.MD}>
                            Button Name
                        </Button>
                        <Button theme={EButtonTheme.SECONDARY} size={EButtonSize.MD}>
                            Button Name
                        </Button>
                    </Page.Header.Title.Controls>
                </Page.Header.Title>
            </Page.Header>
            <Page.Body backgroundColor={backgroundColor}>
                <Row>
                    <Col>
                        <div>
                            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et
                            dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex
                            ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat
                            nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit
                            deserunt mollit anim id est laborum.
                        </div>
                    </Col>
                </Row>
            </Page.Body>
            <Page.Footer>
                <Page.Footer.Description>
                    <Page.Footer.Description.Content>Текст в одну строку</Page.Footer.Description.Content>
                    <Page.Footer.Description.Controls>
                        <Button theme={EButtonTheme.GENERAL} size={EButtonSize.MD}>
                            Button Name
                        </Button>
                        <Button theme={EButtonTheme.SECONDARY} size={EButtonSize.MD}>
                            Button Name
                        </Button>
                    </Page.Footer.Description.Controls>
                </Page.Footer.Description>
            </Page.Footer>
        </Page>
    </div>
</>
```
