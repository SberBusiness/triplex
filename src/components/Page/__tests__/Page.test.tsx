import React from 'react';
import renderer from 'react-test-renderer';
import {Page} from '../Page';

jest.mock('@sberbusiness/icons/SpinnersmallAniIcon20', () => ({
    SpinnersmallAniIcon20: 'svg',
}));

jest.mock('@sberbusiness/icons/SpinnersmallwhiteAniIcon20', () => ({
    SpinnersmallwhiteAniIcon20: 'svg',
}));

jest.mock('@sberbusiness/icons/DropdownwhiteSrvxIcon16', () => ({
    DropdownwhiteSrvxIcon16: 'svg',
}));

jest.mock('@sberbusiness/icons/CaretdownSrvxIcon16', () => ({
    CaretdownSrvxIcon16: 'svg',
}));

jest.mock('@sberbusiness/icons/ClosenotificationSrvxIcon16', () => ({
    ClosenotificationSrvxIcon16: 'svg',
}));

describe.skip('Page', () => {
    beforeEach(() => {
        allure.feature('Page');
    });

    it('renders correctly', () => {
        const tree = renderer
            .create(
                <Page>
                    <Page.Body>Text</Page.Body>
                </Page>
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
