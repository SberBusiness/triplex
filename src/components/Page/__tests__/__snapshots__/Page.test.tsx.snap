// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders correctly 1`] = `
<div
  className="cssClass[page]"
>
  <div
    className="cssClass[body] cssClass[withWhiteBackground]"
    data-markup-guard-id="Body-generateMarkupGuardId"
  >
    <div
      className="cssClass[bodyInner]"
    >
      Text
    </div>
  </div>
</div>
`;
